### DISCUSSION TITLE
SkyWalking gRPC Exporter and traces

### DISCUSSION THREAD

**Discussion by dvisigalli** [open-telemetry/opentelemetry-collector-contrib#7292] [Q&A] [answered]
Will ever SkyWalking gRPC Exporter export traces in SkyWalking native format?
Is already possible, with a specific pipeline configuration, to export traces from OTEL Collector to Skywalking APM?
Thanks

**Comment by jpkrohling** [comment]
cc @liqiangz, @wu-sheng

**Comment by wu-sheng** [comment, marked as answer]
From our previous discussion, this may be not possible. At least, it is nearly impossible.
SkyWalking's trace format includes more information, especially for `parent` part of the RPC entry span, which includes `parent service name`, `parent instance name`, `peer address used in client side`.
If you want to translate the Dapper style span(Zipkin's or Jaeger's are similar), you are building a complex aggregation system, which is not worth it.

The logic behind this is, SkyWalking's trace is designed and built for topology and metrics analysis. So, it sets up all explicit fields and rules when agents build it. 

If you are interested in the `trace to metrics+topology` part, I have to recommend you use our native tracer agents/SDKs.

___
In the meanwhile, exports for metrics and logs are all general adoptive crossing different ecosystems. The exports of those two are existing already.
